6 More Crates

Owai-Seek opened the pull request:
Fixed: Engineering Hardsuit Crate

Removed:
Duplicate Exotic Seed Crate
Festive Wrapping Paper Crate (Replaced)
Fresh Monkey Meat Crate

Tweak:
Grouped some stuff in each tab. (Medical Kits, Space Suits, Ingredients, Meals)
Added Ambrosia Seed to Seed Crate
Added 2 Spray Cans to Biker Gang Crate

Moved:
Grill to Service
All Vending Supplies to a new tab.

Added:
Burger Combo Crate
Cargo Packaging Crate (Replaces Festive Wrap)
Candle Crate
Dirty Magazine Crate (Emag)
Portable Air Pump Crate
Portable Scrubber Crate
~~Pastry Crate~~
~~Re-introduces CBT Interrogation Crate (Contraband)~~
~~Candlelit Dinner Crate~~
~~Pest Control Crate (Contraband)~~

## Why It's Good For The Game

Nobody likes to order things from cargo, maybe this will help.

## Changelog
:cl:
add: Burger, Cargo Packaging, Dirty Magazine, Air Pump, and Scrubber crates.
del: Duplicate Crate, Festive Wrapping Paper Crate, Contraband Monkey Meat Crate
tweak: Gave Seed Crate Ambrosia Seed, gave Biker Gang Crate Spraypaint.
tweak: Organised some crates with sub-categories. Also, moved all vendor refills to a new tab.
tweak: Moved Grill to Service Tab
fix: Engineering Hardsuit Access
/:cl:

If I missed any of my changes let me know, im not trying to hide stuff, im just sleepy.

Owai-Seek committed:
Mmmmm, more crates.

Fixed: Engineering Hardsuit Crate

Removed:
Duplicate Exotic Seed Crate
Festive Wrapping Paper Crate (Replaced)

Tweak:
Grouped some stuff in each tab. (Medical Kits, Space Suits, Ingredients, Meals)
Added Ambrosia Seed to Seed Crate
Added 2 Spray Cans to Biker Gang Crate

Moved:
Grill to Service
All Vending Supplies to a new tab.

Added:
Candle Crate
Dirty Magazine Crate (Emag)
CBT Interrogation Crate (Contraband)
Cargo Packaging Crate
Pest Control Crate (Contraband)
Candlelit Dinner Crate
Portable Air Pump Crate
Portable Scrubber Crate
Pastry Crate
Burger Combo Crate

Known Issues:
I don't know how to get randomised crates to show up in the proper order.

Owai-Seek committed:
Merge branch 'master' into cargocratetglass

Ghommie commented on code/modules/cargo/packs/misc.dm:
```diff
@@ -129,6 +125,37 @@
 //////////////////////////////////////////////////////////////////////////////
 
+/datum/supply_pack/misc/candlelitdinner
+	name = "Candlelit Dinner Crate" //This is meant for greenshifts Ghommie, reeeee.
```
This is just some bob the builder stuff plus wine and candle boxes (ehich have their own pack now).
Remove, I'll reintroduce it in a better fashion later.

Owai-Seek replied:
If you'll fix it later, why can't this be a holdover until that happens?

Ghommie commented on code/modules/cargo/packs/misc.dm:
```diff
@@ -9,20 +9,16 @@
 /datum/supply_pack/misc
 	group = "Miscellaneous Supplies"
 
+/datum/supply_pack/misc/randomised/dirtymags/fill(obj/structure/closet/crate/C)
```
Keep in mind `dirtymags` is the only `misc/randomised` subtype, so the `/randomised` path string is a little superflous. Also move this proc next to its own type.

Owai-Seek replied:
I mean, I don't know how code works, I did that because I was copy/pasting from the other tabs. I suppose I can drop the randomised and stick this underneath the dirtymags box?

Ghommie commented on code/modules/cargo/packs/misc.dm:
```diff
@@ -214,10 +241,48 @@
+	cost = 850
+	contains = list(/obj/item/storage/fancy/candle_box,
+					/obj/item/storage/fancy/candle_box,
+					/obj/item/storage/crayons)
```
We already have artistic supplies with crayons. Remove and add a matchbox instead.

Ghommie commented on code/modules/cargo/packs/misc.dm:
```diff
@@ -214,10 +241,48 @@
+					/obj/item/storage/crayons)
+	crate_name = "candle crate"
+
+/datum/supply_pack/misc/interrogation
```
Listen, just because no one cares about this lousy joke that has no impact on the game, it doesn't mean you got to bring it back.
Remove.

Ghommie commented on code/modules/cargo/packs/service.dm:
```diff
@@ -240,45 +272,17 @@
-	cost = 1500
-	contains = list(/obj/item/vending_refill/cola)
-	crate_name = "soft drinks supply crate"
+/datum/supply_pack/service/janitor/pestcontrol
```
A single bottle of pest killer and some cable coils for three empty grenade casings, some mousetraps, punji sticks, and the bear trap... Actually pretty awful for pest control, feels like you just shoved crap atop of a couple traps as an excuse to add them to the cargo packs.
Remove.

Owai-Seek replied:
It's contraband for a reason, these are antagonist tools. I didnt feel like it should be put in an emag because you can get all of these items fairly easy on-station. People kept noting that there were very few contraband crates.

Ghommie replied:
More than proper antag tools, that's rubbish. Remove.

Ghommie commented on code/modules/cargo/packs/organic.dm:
```diff
@@ -3,18 +3,35 @@
+	crate_name = "combo meal w/toy"
+	crate_type = /obj/structure/closet/crate/wooden
+
+/datum/supply_pack/organic/randomizedcandy
```
Please make the candies a subtype of `/randomized`. And move the num_contained variable to the `/randomized` type.

Ghommie commented on code/modules/cargo/packs/organic.dm:
```diff
@@ -47,51 +64,77 @@
-					/obj/item/reagent_containers/food/snacks/carpmeat,
-					/obj/item/reagent_containers/food/snacks/meat/slab/human)
-	crate_name = "food crate"
+/datum/supply_pack/organic/pastry
```
I mean, we already have the candies crate. Just merge the two together, you sweet tooth.

Owai-Seek replied:
This is meant for birthday parties and such as it gives you several cupcakes and plain cakes (to add stuff to), as well as a reliable way to get donuts and sprinkles from a reliable source.

Ghommie replied:
Just merge the two together, you'll eventually get a couple cupcakes and plain cakes anyway.

Ghommie commented on code/modules/cargo/packs/organic.dm:
```diff
@@ -184,8 +191,35 @@
+					/obj/item/reagent_containers/food/snacks/meat/slab/human)
+	crate_name = "exotic meat crate"
+
+/datum/supply_pack/organic/monkeydripmeat
```
Now, I know this has been around since #9912 (I was not around to give it a quality check), but this is an easy chemistry trick that can be done without the help of cargo in first place, so please remove.

Owai-Seek replied:
People have been asking for more contraband crates, so I've been trying to add more contraband traits. It only shows up if you hack the console, and people who are curious will order it and try to figure out the gimmick.

Ghommie replied:
It's pretty "low quality", simple vague requests do not justify such cheap, meaningless additions. Remove.

Ghommie commented on code/modules/cargo/packs/organic.dm:
```diff
@@ -299,7 +264,12 @@
+	crate_name = "veggie crate"
+
+/datum/supply_pack/organic/randomized/fill(obj/structure/closet/crate/C)
+	for(var/i in 1 to 15)
```
This should become:
```
	for(var/i in 1 to num_contained)
```
After following my suggestion on the randomised candies pack.

Also please move this proc next to the `/organic` parent type, it's shared by many of these supply packs and it's pretty hard to find right now.

Ghommie commented on code/modules/cargo/packs/organic.dm:
```diff
@@ -446,3 +440,11 @@
+	var/list/L = contains.Copy()
+	for(var/i in 1 to num_contained)
+		var/item = pick_n_take(L)
+		new item(C)
```
Aaaand remove this once you repathed the candies supply pack.

On a sidenote: Don't do ever do this again without good reason and keep these niche proc overrides next to their own types, you'll only make reviewing harder this way.

Owai-Seek replied:
To be fair, this is the way it was (or at least, similar to the way it was) before I started messing with it, I only started moving stuff around to try and fix ordering issues.

Owai-Seek commented on code/modules/cargo/packs/misc.dm:
```diff
@@ -214,10 +241,48 @@
+					/obj/item/stock_parts/cell/high)
+	crate_name = "interrogation crate"
+	crate_type = /obj/structure/closet/crate/large
+
```
It's not random, several people have told me they found it funny, and asked to have it moved to contraband because having jokes in emag makes it hard to order.

Owai-Seek committed:
Changed Dirty Mags and Candle

Owai-Seek committed:
Reverts Path changes.

Owai-Seek committed:
Removals.

Removed:
CBT Interrogation
Pastry Crate
Candlelit Dinner Crate
Pest Control Crate

Ghommie committed:
Giving a hand.
